src/components/interpolatedColorLegend.ts

```diff
@@ -104,18 +127,18 @@ export module Components {
     }
 
     public fixedWidth() {
-      return true;
+      return !this.expands() || this._isVertical();
     }
 
     public fixedHeight() {
-      return true;
+      return !this.expands() || !this._isVertical();
     }
 
-    private _generateTicks() {
+    private _generateTicks(numSwatches = InterpolatedColorLegend._DEFAULT_NUM_SWATCHES) {
       let domain = this._scale.domain();
-      let slope = (domain[1] - domain[0]) / this._numSwatches;
+      let slope = (domain[1] - domain[0]) / (numSwatches - 1);
```

Comment:
Doesn't this change the number of swatches generated by default?

Before:
<img width="198" alt="screen shot 2015-08-25 at 9 08 18 pm" src="https://cloud.githubusercontent.com/assets/1440449/9485399/7f65b804-4b6d-11e5-92f3-932b46fd792c.png">

After:
<img width="186" alt="screen shot 2015-08-25 at 9 08 09 pm" src="https://cloud.githubusercontent.com/assets/1440449/9485408/86068918-4b6d-11e5-846a-1a0ea56eedbd.png">

On the other hand, seeing as the previous logic generated `this._numSwatches + 1` swatches, this is probably fine...